integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
+    }
+
+    /**
+     * This method will set the source row count if it does not already exists. This happens when
+     * other processors are used to pull data, such as ImportSqoop
+     * 
+     * If ImportSqoop Nifi processor is used, it updates the source.row.count value with the sqoop
+     * count
+     */
+    protected void setSourceRowCount() {
+        try {
+            if (flowAttributes.containsAttribute(DataQualityConstants.SQOOP_ROW_COUNT_ATTRIBUTE)) {
+                String sqoopRowCount = flowAttributes.getAttributeValue(DataQualityConstants.SQOOP_ROW_COUNT_ATTRIBUTE);
+                flowAttributes.addAttribute(DataQualityConstants.SOURCE_ROW_COUNT_ATTRIBUTE, sqoopRowCount);
+            }
+        } catch (MissingAttributeException e) {
+            log.error("Required Attribute missing");
+        }
+    }
+
+    /**
+     * Given the passed in arguments, calculates the row counts
+     * 
+     * @param databaseName Name of the Hive database
+     * @param tableName Name of the Hive table
+     * @param whereClause Additional filters
+     * @return row count for the table
+     */
+    protected long getRowCount(String databaseName, String tableName, String whereClause) {
+        try {
+
+            SparkContext sparkContext = SparkContext.getOrCreate();
+            hiveContext = new org.apache.spark.sql.hive.HiveContext(sparkContext);
+
+            String query = "SELECT COUNT(*) FROM " + databaseName
+                           + "."
+                           + tableName
+                           + " ";
+
+            if (whereClause != "") {
+                query = query + "WHERE " + whereClause;
+            }
+
+            log.info("Executing hive query: " + query);
+
+            DataFrame countDF = hiveContext.sql(query);
+
+            // Only take the first value which contains the row count
+            Long rowCount = countDF.collect()[0].getLong(0);
+
+            return rowCount;
+
+        } catch (Exception e) {
+            log.error("ERROR - Error while getting row count. Parameters were " +
+                      " database = "
+                      + databaseName
+                      +
+                      " table = "
+                      + tableName
+                      +
+                      " whereClause = "
+                      + whereClause, e);
+
+            throw e;
+        }
     }
 
-    public void doDataQualityChecks() {
+    /**
+     * Write the results to a Hive table
+     */
+    private void outputToHive() {
+
         try {
+            String databaseName = flowAttributes.getAttributeValue(DataQualityConstants.CATEGORY_ATTRIBUTE);
+            String tableName = flowAttributes.getAttributeValue(DataQualityConstants.FEED_ATTRIBUTE);
+            String dqTableName = tableName + DataQualityConstants.DQ_TABLE_SUFFIX;
+            String processing_dttm = flowAttributes.getAttributeValue(DataQualityConstants.PROCESSING_DTTM_ATTRIBUTE);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00117)